### PR TITLE
added my webskype project

### PR DESCRIPTION
**Pull request by Nastia-Kv**
Please review my project.

### CODE REVIEW

**Comment by Nastia-Kv** [comment]
@khospodarysko

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+        <!--</dependency>-->
+
+        <dependency>
+            <groupId>org.testng</groupId>
+            <artifactId>testng</artifactId>
+            <version>6.10</version>
+            <scope>test</scope>
+        </dependency>
+
+        <dependency>
```
Comment (kostia-hospodarysko) [inline pom.xml]: selenium-java has all those dependencies already, those below can be removed.

File: src/main/java/skype/BasePage.java [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+import java.util.concurrent.TimeUnit;
+
+import static utils.Timeouts.DEFAULT_WAIT_SECONDS;
+
+
+/**
+ * Created by nastia on 2/26/17.
+ */
+public class BasePage {
+    public String PAGE_URL;
```
Comment (kostia-hospodarysko) [inline src/main/java/skype/BasePage.java]: Should be camelCase for such variables, convention is to use all capital for constants.

File: src/main/java/skype/BasePage.java [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+    public String PAGE_TITLE;
+    public WebDriver driver;
+
+
+    public BasePage(WebDriver driver) {
+        PageFactory.initElements(driver, this);
+        this.driver = driver;
+
```
Comment (kostia-hospodarysko) [inline src/main/java/skype/BasePage.java]: Redundant empty lines. Correct all over the code.

File: src/main/java/skype/BasePage.java [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+    }
+
+    //basic methods
+    public void loadPage() {
+        driver.get(getPageUrl());
+        Assert.assertEquals(driver.getTitle(), PAGE_TITLE);
```
Comment (kostia-hospodarysko) [inline src/main/java/skype/BasePage.java]: General practice is to use asserts only in tests.

File: src/main/java/skype/BasePage.java [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+            } catch (NoSuchElementException e) {
+                System.out.println("NO element in DOM");
+            }
+        }
+        System.out.println("Time  " + (System.currentTimeMillis() - start));
+        driver.manage().timeouts().implicitlyWait(DEFAULT_WAIT_SECONDS, TimeUnit.SECONDS);
+        return false;
+    }
+
+    public void waitForElementToBeDisplayed2(WebElement element) {
```
Comment (kostia-hospodarysko) [inline src/main/java/skype/BasePage.java]: These two are the same, only time differs, one could be deleted.

File: src/main/java/skype/call_phones_pages/CallPhonesPage.java [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+//            System.out.println(element.getText());
+//        }
+
+        //or
+
+        for (WebElement element : listBox2){
+            System.out.println(element.getText());
+        }
+
+        for (WebElement element : listBox2){
```
Comment (kostia-hospodarysko) [inline src/main/java/skype/call_phones_pages/CallPhonesPage.java]: Strange logic, why to click on the same countryUkraine a few times in a loop?

File: src/main/java/skype/call_phones_pages/CallPhonesPage.java [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+
+
+    public void makeCall() throws InterruptedException {
+        enterNumberField.sendKeys("0680077626");
+//        WebDriverWait webDriverWait = new WebDriverWait(driver, 60);
+//        webDriverWait.until(ExpectedConditions.stalenessOf(chooseCountryDropdown));
+//        webDriverWait.until(ExpectedConditions.presenceOfElementLocated(By.cssSelector(".SelectBox-header")));
+
+//
+        if (!isVisibleWithTryCatch(chooseCountryDropdown, 1000000))
```
Comment (kostia-hospodarysko) [inline src/main/java/skype/call_phones_pages/CallPhonesPage.java]: Seems a very big timeout, could it be less, like 10 seconds?

File: src/main/java/skype/contacts_pages/ContactsPage.java [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+
+    public int countAllContacts() {
+        WebDriverWait webDriverWait = new WebDriverWait(driver, 60);
+        webDriverWait.until(ExpectedConditions.visibilityOf(soundTestService));
+        int contactsNumber = 0;
+        for (WebElement contact : allContacts) {
+            contactsNumber += 1;
+        }
+        System.out.println(contactsNumber);
+        return contactsNumber;
```
Comment (kostia-hospodarysko) [inline src/main/java/skype/contacts_pages/ContactsPage.java]: Could use allContacts.size();

File: src/main/java/skype/contacts_pages/ContactsPage.java [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+    }
+
+    public void selectKContact() throws InterruptedException {
+        WebDriverWait webDriverWait = new WebDriverWait(driver, 60);
+        webDriverWait.until(ExpectedConditions.visibilityOf(soundTestService));
+
+        int n = 0;
+        for (WebElement element : namesKContacts) {
```
Comment (kostia-hospodarysko) [inline src/main/java/skype/contacts_pages/ContactsPage.java]: Could use ```WebElement element = namesKContacts.get(0)``` returns 1st element.

File: src/main/java/skype/home_pages/HomePage.java [hunk 10]
```diff
@@ -0,0 +1,308 @@
...
+//            System.out.println("Exception");
+//        }
+    }
+
+    public void searchForExistingNewContact2(String text) throws InterruptedException {
+        waitForElementToBeDisplayed(searchSkypeField);
+        click(searchSkypeField);
+        Actions actions = new Actions(driver);
+        actions.moveToElement(focusedSearchSkypeField).click().sendKeys(text).build().perform();
+        WebDriverWait webDriverWait = new WebDriverWait(driver, 60);
```
Comment (kostia-hospodarysko) [inline src/main/java/skype/home_pages/HomePage.java]: These two lines are used a lot, could be moved to some utility method, and be invoked like ```waitForElementToBeDisplayed(searchSkypeField);```.

File: src/main/java/skype/home_pages/HomePage.java [hunk 11]
```diff
@@ -0,0 +1,308 @@
...
+    public void searchForContactInContactList2(String text) throws InterruptedException {
+        waitForElementToBeDisplayed(searchSkypeField);
+        click(searchSkypeField);
+        Actions actions = new Actions(driver);
+        actions.moveToElement(focusedSearchSkypeField).click().sendKeys(text).build().perform();
+        WebDriverWait webDriverWait = new WebDriverWait(driver, 60);
+        webDriverWait.until(ExpectedConditions.visibilityOf(searchSkypeDirectoryBtn2));
+        actions.moveToElement(searchSkypeDirectoryBtn2).click().build().perform();
+
+        try {
```
Comment (kostia-hospodarysko) [inline src/main/java/skype/home_pages/HomePage.java]: Such try-catches could be also refactored to some utility method.

**Pull request merged**